config/v1/types_infrastructure.go

```diff
@@ -478,6 +478,35 @@ type AzurePlatformStatus struct {
 	// armEndpoint specifies a URL to use for resource management in non-soverign clouds such as Azure Stack.
 	// +optional
 	ARMEndpoint string `json:"armEndpoint,omitempty"`
+
+	// resourceTags is a list of additional tags to apply to Azure resources created for the cluster.
+	// See https://docs.microsoft.com/en-us/rest/api/resources/tags for information on tagging Azure resources.
+	// Due to limitations on Automation, Content Delivery Network, DNS Azure resources, a maximum of 15 tags
+	// may be applied. OpenShift reserves 5 tags for internal use, allowing 10 tags for user configuration.
+	// +kubebuilder:validation:MaxItems=10
+	// +optional
+	ResourceTags []AzureResourceTag `json:"resourceTags,omitempty"`
+}
+
+// AzureResourceTag is a tag to apply to Azure resources created for the cluster.
+type AzureResourceTag struct {
+	// key is the key part of the tag. A tag key can have a maximum of 128 characters and cannot be empty. Key
+	// must begin with a letter, end with a letter, number or underscore, and can contain only letters, numbers,
+	// underscores, periods, or hyphens.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z]([0-9A-Za-z_.-]*[0-9A-Za-z_])?$`
+	// +required
+	Key string `json:"key"`
+	// value is the value part of the tag. A tag value can have a maximum of 256 characters and cannot be empty. Value
+	// can contain only `a-zA-Z0-9_.=+-@` characters.
```

Comment:
Can we spell out the allow characters in prose as we have on the key field?

Reply:
Where did this validation come from? Is there a link to Azure that details this validation? I'd like to review how they document it before we change our validation.

We could perhaps word it as
```
The value must contain only alphanumeric characters and the following special characters `_ + , - . / : ; < = > ? @`.
```

Reply:
This [link](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/tag-resources?tabs=json#limitations) on tag limitations mention restricted characters for tag names and for values it's mentioned as `The value can contain all characters.`
Though it says value can contain all characters, DNS resource value can't contain charcters `# $ % ^` found when testing the allowed characters, hence restricted character set for value as well.

Reply:
Ok yep, are you happy with my suggested wording from the previous comment then?